pkg/component/gardener/dashboard/terminal/deployment.go

```diff
@@ -0,0 +1,183 @@
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot: ptr.To(true),
+						RunAsUser:    ptr.To[int64](65532),
+					},
+					TerminationGracePeriodSeconds: ptr.To(int64(10)),
+					Containers: []corev1.Container{{
+						Name:            name,
+						Image:           t.values.Image,
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Args: []string{
+							"--config-file=" + volumeMountPathConfig + "/" + dataKeyConfig,
+							"--zap-devel=false",
+						},
+						Env: []corev1.EnvVar{{
+							Name:  "KUBECONFIG",
+							Value: gardenerutils.PathGenericKubeconfig,
+						}},
+						Resources: corev1.ResourceRequirements{
+							Requests: map[corev1.ResourceName]resource.Quantity{
+								corev1.ResourceCPU:    resource.MustParse("10m"),
+								corev1.ResourceMemory: resource.MustParse("50Mi"),
+							},
+						},
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          portNameAdmission,
+								ContainerPort: portAdmission,
+								Protocol:      corev1.ProtocolTCP,
+							},
+							{
+								Name:          portNameMetrics,
+								ContainerPort: portMetrics,
+								Protocol:      corev1.ProtocolTCP,
+							},
+						},
+						LivenessProbe: &corev1.Probe{
+							ProbeHandler: corev1.ProbeHandler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path:   "/healthz",
+									Port:   intstr.FromInt32(portProbes),
+									Scheme: corev1.URISchemeHTTP,
+								},
+							},
+							InitialDelaySeconds: 15,
+							TimeoutSeconds:      5,
+							FailureThreshold:    6,
+							SuccessThreshold:    1,
+							PeriodSeconds:       20,
+						},
+						ReadinessProbe: &corev1.Probe{
+							ProbeHandler: corev1.ProbeHandler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path:   "/readyz",
+									Port:   intstr.FromInt32(portProbes),
+									Scheme: corev1.URISchemeHTTP,
+								},
+							},
+							InitialDelaySeconds: 5,
+							TimeoutSeconds:      5,
+							FailureThreshold:    6,
+							SuccessThreshold:    1,
+							PeriodSeconds:       10,
+						},
+						SecurityContext: &corev1.SecurityContext{AllowPrivilegeEscalation: ptr.To(false)},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      volumeNameConfig,
+								MountPath: volumeMountPathConfig,
+							},
+							{
+								Name:      volumeNameServerCert,
+								MountPath: "/tmp/k8s-webhook-server/serving-certs",
+								ReadOnly:  true,
+							},
+							{
+								Name:      volumeNameServerCert,
+								MountPath: "/tmp/k8s-metrics-server/serving-certs",
+								ReadOnly:  true,
+							},
+						},
+					}},
+					Volumes: []corev1.Volume{
+						{
+							Name: volumeNameConfig,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{Name: configMapName},
+									Items: []corev1.KeyToPath{{
+										Key:  dataKeyConfig,
+										Path: dataKeyConfig,
+									}},
```

Comment:
I know this is the same also in the TCM helm chart but I think we can drop `Items` for simplicity, or do you think that makes it more explicit?